src/test/resources/tpcds/spark-2.4/approved-plans-v1_4/q13/simplified.txt

```diff
@@ -1,57 +1,49 @@
+WholeStageCodegen
+  HashAggregate [avg(UnscaledValue(ss_ext_sales_price)),avg(UnscaledValue(ss_ext_wholesale_cost)),avg(cast(ss_quantity as bigint)),count,count,count,sum,sum,sum,sum,sum(UnscaledValue(ss_ext_wholesale_cost))] [avg(UnscaledValue(ss_ext_sales_price)),avg(UnscaledValue(ss_ext_wholesale_cost)),avg(cast(ss_quantity as bigint)),avg(ss_ext_sales_price),avg(ss_ext_wholesale_cost),avg(ss_quantity),count,count,count,sum,sum,sum,sum,sum(UnscaledValue(ss_ext_wholesale_cost)),sum(ss_ext_wholesale_cost)]
     InputAdapter
       Exchange #1
-        WholeStageCodegen (6)
-          HashAggregate [ss_quantity,ss_ext_sales_price,ss_ext_wholesale_cost] [sum,count,sum,count,sum,count,sum,sum,count,sum,count,sum,count,sum]
-            Project [ss_quantity,ss_ext_sales_price,ss_ext_wholesale_cost]
-              BroadcastHashJoin [ss_hdemo_sk,hd_demo_sk,cd_marital_status,cd_education_status,ss_sales_price,hd_dep_count]
-                Project [ss_hdemo_sk,ss_quantity,ss_sales_price,ss_ext_sales_price,ss_ext_wholesale_cost,cd_marital_status,cd_education_status]
-                  BroadcastHashJoin [ss_cdemo_sk,cd_demo_sk,cd_marital_status,cd_education_status,ss_sales_price]
-                    Project [ss_cdemo_sk,ss_hdemo_sk,ss_quantity,ss_sales_price,ss_ext_sales_price,ss_ext_wholesale_cost]
-                      BroadcastHashJoin [ss_sold_date_sk,d_date_sk]
-                        Project [ss_sold_date_sk,ss_cdemo_sk,ss_hdemo_sk,ss_quantity,ss_sales_price,ss_ext_sales_price,ss_ext_wholesale_cost]
-                          BroadcastHashJoin [ss_addr_sk,ca_address_sk,ca_state,ss_net_profit]
-                            Project [ss_sold_date_sk,ss_cdemo_sk,ss_hdemo_sk,ss_addr_sk,ss_quantity,ss_sales_price,ss_ext_sales_price,ss_ext_wholesale_cost,ss_net_profit]
-                              BroadcastHashJoin [ss_store_sk,s_store_sk]
-                                Filter [ss_store_sk,ss_addr_sk,ss_sold_date_sk,ss_cdemo_sk,ss_hdemo_sk,ss_net_profit,ss_sales_price]
-                                  ColumnarToRow
-                                    InputAdapter
-                                      Scan parquet default.store_sales [ss_sold_date_sk,ss_cdemo_sk,ss_hdemo_sk,ss_addr_sk,ss_store_sk,ss_quantity,ss_sales_price,ss_ext_sales_price,ss_ext_wholesale_cost,ss_net_profit]
+        WholeStageCodegen
+          HashAggregate [count,count,count,count,count,count,ss_ext_sales_price,ss_ext_wholesale_cost,ss_quantity,sum,sum,sum,sum,sum,sum,sum,sum] [count,count,count,count,count,count,sum,sum,sum,sum,sum,sum,sum,sum]
+            Project [ss_ext_sales_price,ss_ext_wholesale_cost,ss_quantity]
+              BroadcastHashJoin [cd_education_status,cd_marital_status,hd_demo_sk,hd_dep_count,ss_hdemo_sk,ss_sales_price]
+                Project [cd_education_status,cd_marital_status,ss_ext_sales_price,ss_ext_wholesale_cost,ss_hdemo_sk,ss_quantity,ss_sales_price]
+                  BroadcastHashJoin [cd_demo_sk,ss_cdemo_sk]
```

Comment:
check this part?

Reply:
maybe because the remaining columns are used in higher level broadcast join two lines above

```
BroadcastHashJoin [cd_education_status,cd_marital_status,hd_demo_sk,hd_dep_count,ss_hdemo_sk,ss_sales_price]
```